File: 20-valid-parentheses/valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stk = new Stack<>();
        if(s.length() % 2 != 0) return false;
        for(char ch : s.toCharArray()){
            if(ch == '(' || ch == '{' || ch == '[')stk.push(ch);
            else if(!stk.isEmpty() && match(stk.peek(),ch)) {
                stk.pop();
            }
            else return false;
        }
        return stk.isEmpty();
    }
    boolean match(char ch1, char ch2){
        if(ch1 == '(' && ch2 == ')') return true;
        if(ch1 == '{' && ch2 == '}') return true;
        if(ch1 == '[' && ch2 == ']') return true;
        return false;
    }
}

File: 20-valid-parentheses/valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        lst=[]
        if len(s)%2==0:
            for i in range(len(s)):
                if s[i]=="[" or s[i]=="{" or s[i]=="(" :
                    lst.append(s[i])
                else:
                    if lst and self.check(lst[-1], s[i]):
                        lst.pop() 
                    else:
                        return False
        else:
            return False
        if not lst:
            return True
        else:
            return False
    def check(self,ch1,ch2):
        return (ch1 == "(" and ch2 == ")") or (ch1 == "{" and ch2 == "}") or (ch1 == "[" and ch2 == ']')